File: scala/75.scala
object Solution {
    def sortColors(nums: Array[Int]): Unit = {
        var red = 0
        var white = 0
        var blue = nums.length - 1

        while (white <= blue) {
            nums(white) match {
                case 0 =>
                    swap(nums, red, white)
                    red += 1
                    white += 1
                case 1 =>
                    white += 1
                case 2 =>
                    swap(nums, white, blue)
                    blue -= 1
            }
        }
    }
    
    def swap(nums: Array[Int], i: Int, j: Int): Unit = {
        val temp = nums(i)
        nums(i) = nums(j)
        nums(j) = temp
    }
}

File: py/75.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        red, blue = 0, n - 1
        idx = 0
        while idx <= blue:
            if nums[idx] == 0:
                nums[idx], nums[red] = nums[red], nums[idx]
                red += 1
            elif nums[idx] == 2:
                nums[idx], nums[blue] = nums[blue], nums[idx]
                blue -= 1
            else:
                idx += 1
